src/Framework/IBuildEngine6.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// This interface extends <see cref="IBuildEngine5" /> to allow tasks to get the current project's global properties.
+    /// </summary>
+    public interface IBuildEngine6 : IBuildEngine5
+    {
+        /// <summary>
+        /// Gets the global properties for the current project.
+        /// </summary>
+        /// <returns>A <see cref="Dictionary{String, String}" /> containing the global properties of the current project.</returns>
+        Dictionary<string, string> GetGlobalProperties();
```

Comment:
This should probably return `IDictionary<TKey, TValue>`.  I couldn't return `IReadOnlyDictionary<TKey, TValue>` since its not in .NET 3.5

Reply:
You can wrap the implementation like the current ones:

https://github.com/microsoft/msbuild/blob/74c08313395ee87d58ef53f8a7918a937082be01/src/MSBuild/OutOfProcTaskHostNode.cs#L380-L381